### ISSUE TITLE
UserWarning: Passing parameter(s) to COUNT is deprecated.

### ISSUE THREAD

**Issue by znation** [apple/turicreate#2529]
I get this warning while following the image classifier chapter of the user guide at https://apple.github.io/turicreate/docs/userguide/image_classifier/:

```python
# Evaluate the model and print the results
metrics = model.evaluate(test_data)
```

```
| 4800             | 5m 26s       | 97.25%           |
| 4864             | 5m 30s       | 98.5%            |
| 4880             | 5m 31s       | 100%             |
+------------------+--------------+------------------+
/Users/zach/venv/lib/python2.7/site-packages/turicreate/aggregate.py:92: UserWarning: Passing parameter(s) to COUNT is deprecated. This functionality will be removed in the next major release. Any passed parameters are ignored.
  + 'the next major release. Any passed parameters are ignored.')
```

At no point did I actually call this "deprecated" API directly. We should update our own internal calls to make sure we don't give warnings to users while calling non-deprecated API.

**Comment by znation** [comment]
This is on 5.8.

**Comment by TobyRoseman** [comment]
This is my fault. It's a bug in determining if we should omit a warning. [This line](https://github.com/apple/turicreate/blob/e520ad097c8a565c710660169cf95719335147dc/src/python/turicreate/aggregate.py#L90) should be `if len(args) != 0:` rather than `if len(args) == 0:`.

**Comment by TobyRoseman** [comment]
Fixed by #2579
